src/store/SettingsDialogStore.js

```diff
@@ -7,18 +7,15 @@ export default class SettingsDialogStore {
     @observable items = []; // [{id: '', title: '', value: ''}]
     @observable title = '';
     @observable description = '';
-    @observable id;
 
     @observable activeTab = 'settings'; // 'settings' | 'description'
     @computed get showTabs() { return !!this.description; }
 
     constructor({
-        mainStore, getContext, onDeleted, favoritesId, onChanged,
+        mainStore, getContext, onChanged,
```

Comment:
@shayan-binary IMO, we should keep the delete button in the setting dialog, since the default way of opening the setting page is by right clicking on the drown object. I'll talk about it with UI/UX team.